subprojects/model-core/src/main/java/org/gradle/internal/instantiation/ManagedTypeFactory.java

```diff
@@ -16,18 +16,23 @@
 
 package org.gradle.internal.instantiation;
 
+import com.google.common.base.Objects;
 import org.gradle.api.reflect.ObjectInstantiationException;
-import org.gradle.internal.state.Managed;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.state.ManagedFactory;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
 
-public class ManagedTypeFactory implements Managed.Factory {
+public class ManagedTypeFactory implements ManagedFactory {
     private final Constructor<?> constructor;
 
-    // Used by generated code
-    public ManagedTypeFactory(Class<?> type) throws NoSuchMethodException {
-        constructor = type.getConstructor(Object[].class);
+    public ManagedTypeFactory(Class<?> type) {
+        try {
+            constructor = type.getConstructor(Object[].class);
+        } catch (NoSuchMethodException e) {
+            throw UncheckedException.throwAsUncheckedException(e);
```

Comment:
should we show a better message here?

Reply:
It's probably not super important since this shouldn't be used outside of generated managed types (which should always have the right constructor) but providing a better message wouldn't hurt.  Done.